control-plane/config-entries/controllersv2/mesh_gateway_controller.go

```diff
@@ -126,7 +125,67 @@ func (r *MeshGatewayController) onCreateUpdate(ctx context.Context, req ctrl.Req
 		return fmt.Errorf("unable to create role binding: %w", err)
 	}
 
-	// TODO NET-6393
+	//Create Service
+
+	areServicesEqual := func(a, b *corev1.Service) bool {
+
+		if a == nil || b == nil {
+			return true
```

Comment:
should this be false if either is nil? like if a exists and b is nil then the would not be equal?

Reply:
Yes, if "a" exists, and "b" is nil then service would not be equal. The idea here was if service "a"  or "b" is nil, then there really isn't a need to merge from nil to an existing service. But after re-reading the `mergeServiceOp` function, we do check that usecase and return an error. So I'll return false here. Good catch!

Reply:
do we care if both a and b are nil?

Reply:
We do! Technically (I think), service "b" or builtService object should never be nil as that's the service we want to create/update.